src/libUtils/SafeMath.h

```diff
@@ -41,6 +41,55 @@ class SafeMath {
     return true;
   }
 
+  static bool pow(const T& a, const T& b, T& result) {
+    if (b == 0) {
+      result = 1;
+      return true;
+    }
+
+    if (b < 0) {
+      LOG_GENERAL(WARNING, "Doesn't support pow with negative index");
+      return false;
+    }
+
+    T temp = a, count = b;
+
+    while (count > 0) {
+      if (!SafeMath::mul(temp, a, temp)) {
+        LOG_GENERAL(WARNING, "SafeMath::pow failed");
+        return false;
+      }
+      --count;
+    }
+
+    result = temp;
+    return true;
+  }
+
+  // be careful when using
```

Comment:
can describe more?

Reply:
I think we can just keep one `pow` and have an optional `bool isCritical = true` parameter?

Reply:
Maybe describe the use case in the inline comments